layers/layer_options.cpp

```diff
@@ -329,12 +329,63 @@ static bool SetBool(std::string &config_string, std::string &env_string, bool de
     return result;
 }
 
+static std::string GetConfigValue(const char *setting) { return getLayerOption((std::string("khronos_validation.") + setting).c_str()); }
+
+static std::string GetEnvVarValue(const char *setting) {
+    std::string env_var = setting;
+    std::transform(env_var.begin(), env_var.end(), env_var.begin(), ::toupper);
+    return GetEnvironment((std::string("VK_LAYER_") + env_var).c_str());
+}
+
+static std::optional<std::string> GetSettingValue(const char *setting) {
+    const std::string cfg_value = GetConfigValue(setting);
+    const std::string env_value = GetEnvVarValue(setting);
+
+    if (!env_value.empty()) {
+        return env_value;
+    } else if (!cfg_value.empty()) {
+        return cfg_value;
+    }
+    return {};
+}
+
+static void SetValidationSetting(CHECK_DISABLED &disable_data, const DisableFlags feature_disable, const char *setting) {
+    const std::optional<std::string> setting_value = GetSettingValue(setting);
+
+    if (setting_value) {
+        disable_data[feature_disable] = setting_value != "true";
+    }
+}
+
+static void SetValidationSetting(CHECK_ENABLED &enable_data, const EnableFlags feature_enable, const char *setting) {
```

Comment:
FFTI

Instead of having a `SetValidationSettings` function for `DisableFlags`/`EnableFlags` what about 
```suggestion
static void SetValidationSetting(bool * feature, const char *setting) {
```

And at the call sight it would look like this:

```c++
SetValidationSetting(&settings_data->enables[best_practices], "validate_best_practices");
...
SetValidationSetting(&settings_data->disables[stateless_checks], "stateless_param");
```

Reply:
:heavy_plus_sign: for this, though it should be `bool&` instead of `bool*` (https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs).

Reply:
Except that the comparison operator is not the same in the function so I am not really convinced of the value of this change.